### PR TITLE
Support ruby 2.6 Hash#merge with multiple arguments

### PR DESCRIPTION
**Pull request by iguchi1124**
close https://github.com/sinatra/sinatra/issues/1571

I tested locally:
```
➜  sinatra bundle exec ruby test/indifferent_hash_test.rb
Run options: --seed 9474

# Running:

...........................

Finished in 0.004303s, 6274.6921 runs/s, 23471.9963 assertions/s.

27 runs, 101 assertions, 0 failures, 0 errors, 0 skips
```

### CODE REVIEW

**Comment by iguchi1124** [comment]
Test failing from this changes https://github.com/rack/rack/pull/1391 :eyes:

**Comment by jkowens** [comment]
@iguchi1124 thanks for spotting the rack issue. I reported the bug and it was fixed there.

**Comment by iguchi1124** [comment]
> I reported the bug and it was fixed there.

thanks!

File: test/indifferent_hash_test.rb [hunk 1]
```diff
@@ -205,6 +205,21 @@ def test_merge
     assert_equal 2, hash2[?q]
   end
 
+  def test_merge_with_multiple_argument
+    hash = Sinatra::IndifferentHash.new.merge({a: 1}, {b: 2}, {c: 3})
+    assert_equal 1, hash[?a]
+    assert_equal 2, hash[?b]
+    assert_equal 3, hash[?c]
+    assert_equal 3, hash[?c]
```
Comment (jkowens) [inline test/indifferent_hash_test.rb]: This line appears to be a duplicate.

File: lib/sinatra/indifferent_hash.rb [hunk 2]
```diff
@@ -132,22 +132,26 @@ def values_at(*keys)
...
-      return super if other_hash.is_a?(self.class)
-
-      other_hash.each_pair do |key, value|
-        key = convert_key(key)
-        value = yield(key, self[key], value) if block_given? && key?(key)
-        self[key] = convert_value(value)
+    def merge!(*other_hashes)
+      other_hashes.each do |other_hash|
+        if other_hash.is_a?(self.class)
+          super
```
[the remark this answers is not in the document]
Comment (iguchi1124) [reply lib/sinatra/indifferent_hash.rb]: https://github.com/sinatra/sinatra/pull/1572/commits/6f46895ebfbb181a17b80eb8629b6dae301b02b4

**Comment by jkowens** [comment]
LGTM 👍 

@namusyaka what do you think?

**Comment by iguchi1124** [comment]
Hi, I checked [`cf1c6b1`](https://github.com/sinatra/sinatra/blob/cf1c6b1/lib/sinatra/indifferent_hash.rb) code which is released as [v2.0.8](https://github.com/sinatra/sinatra/blob/v2.0.8/lib/sinatra/indifferent_hash.rb), but these changes seems disappear.  Could you check push logs? :pray:

@namusyaka

**Comment by namusyaka** [comment]
Unfortunately, I can't see the push log because of the permission issue.
However, I just found the force-push message in [this PR](https://github.com/sinatra/sinatra/pull/1584#event-2916403767).
According to the message, I force-pushed the wrong commit revision in the master branch.

Based on the above assumption, I'm checking if there is other commit that was missing, but it's not found except this.
So, I'm going to re-release v2.0.8.1 for making this commit included in the version.

I'm sorry for my mistake on the release operation.

**Comment by namusyaka** [comment]
Released: https://rubygems.org/gems/sinatra/versions/2.0.8.1
I confirmed that the released gem contains this change correctly.

Again, I was making a completely crazy mistake. Thanks so much for letting me know. And I'm really sorry.

**Pull request merged**